packages/flutter/lib/src/widgets/text_editing_intents.dart

```diff
@@ -221,12 +221,25 @@ class ExtendSelectionVerticallyToAdjacentPageIntent extends DirectionalCaretMove
   }) : super(forward, collapseSelection);
 }
 
+/// Extends, or moves the current selection from the current
+/// [TextSelection.extent] position to the previous or the next paragraph
+/// boundary.
```

Comment:
I feel "paragraph boundary" could be harder to understand to most readers than something like "start of the next paragraph", or "end of the current paragraph".